code/model/DNEnvironment.php

```diff
@@ -614,13 +614,24 @@ public function CurrentBuild() {
 	}
 
 	/**
-	 * A history of all builds deployed to this environment
-	 *
+	 * A list of past deployments.
 	 * @return ArrayList
 	 */
 	public function DeployHistory() {
 		return $this->Deployments()
-			->where('SHA IS NOT NULL')
+			->where('"SHA" IS NOT NULL')
+			->filter('State', ['Completed', 'Failed', 'Invalid'])
```

Comment:
I just dislike magic numbers and strings :) I have fat fingers and it's easy to miss uppercase / lowercase as well. :P